hyperspy/io_plugins/impulse.py

```diff
@@ -0,0 +1,207 @@
+            .replace("°C", "C")
+            .replace("#", "No")
+            .replace("(", "")
+            .replace(")", "")
+            .replace("/", "_")
+            .replace("%", "Perc")
+            for name in self.column_names
+        ]
+        data = np.genfromtxt(
+            self.filename,
+            delimiter=",",
+            dtype=None,
+            names=names,
+            skip_header=1,
+            encoding="latin1",
+        )
+        self._data_dictionary = dict()
+        for i, name, name_dtype in zip(range(len(names)), self.column_names, names):
+            if name == "Experiment time":
+                self.time_axis = data[name_dtype]
+            elif name == "MixValve":
+                mixvalvedatachanged = data[name_dtype]
+                for index, item in enumerate(data[name_dtype]):
+                    mixvalvedatachanged[index] = (
+                        int(int(item.split(";")[0]) + 2) * 100
+                        + (int(item.split(";")[1]) + 2) * 10
+                        + (int(item.split(";")[2]) + 2)
+                    )
+                mixvalvedatachangedint = np.array(mixvalvedatachanged, dtype=np.int32)
+                self._data_dictionary[name] = mixvalvedatachangedint
+            else:
+                self._data_dictionary[name] = data[name_dtype]
+
+    def _read_metadatafile(self):
+        # Locate the experiment metadata file
+        self.original_metadata = {}
+        notes = []
+        notes_section = False
+
+        if "_Synchronized data" in str(self.filename) or "raw" in str(
+            self.filename
+        ):  # Check if Impulse filename formatting is intact
+            metadata_file = (
+                "_".join(str(self.filename).split("_")[:-1]) + "_Metadata.log"
+            ).replace("\\", "/")
+            if os.path.isfile(metadata_file):
+                with open(metadata_file, newline="") as csvfile:
+                    metadata_file_reader = csv.reader(csvfile, delimiter=",")
+                    for row in metadata_file_reader:
+                        if notes_section:
+                            notes.append(row[0])
+                        elif row[0] == "Live notes":
+                            notes_section = True
+                            notes = [row[1].strip()]
+                        else:
+                            self.original_metadata[row[0].replace(" ", "_")] = row[
+                                1
+                            ].strip()
+                    self.original_metadata["Notes"] = notes
+                    self.start_datetime = np.datetime64(
+                        dt.strptime(
+                            self.original_metadata["Experiment_date"]
+                            + self.original_metadata["Experiment_time"],
+                            "%d-%m-%Y%H:%M:%S",
+                        )
+                    )
+
+            else:
+                raise IOError(no_metadata_file_error)
```

Comment:
Even if there is no metadata file or it is corrupted, it would still be useful to be able to open the file? Maybe simply add a warning?